pkg/apiserver/event/event_test.go

```diff
@@ -0,0 +1,335 @@
+			CreatedAt:    time.Time{},
+			UpdatedAt:    time.Time{},
+			DeletedAt:    nil,
+			Experiment:   "testExperiment",
+			Namespace:    "testNamespace",
+			Kind:         "testKind",
+			Message:      "test",
+			StartTime:    nil,
+			FinishTime:   nil,
+			Duration:     "1m",
+			Pods:         nil,
+			ExperimentID: "testUID",
+		}
+		res = append(res, event)
+	} else {
+		err = fmt.Errorf("test err")
+	}
+	return res, err
+}
+
+func (m *MockEventService) DryListByFilter(_ context.Context, filter core.Filter) ([]*core.Event, error) {
+	var res []*core.Event
+	var err error
+	if filter.Kind == "testKind" {
+		event := &core.Event{
+			ID:           0,
+			CreatedAt:    time.Time{},
+			UpdatedAt:    time.Time{},
+			DeletedAt:    nil,
+			Experiment:   "testExperiment",
+			Namespace:    "testNamespace",
+			Kind:         "testKind",
+			Message:      "test",
+			StartTime:    nil,
+			FinishTime:   nil,
+			Duration:     "1m",
+			Pods:         nil,
+			ExperimentID: "testUID",
+		}
+		res = append(res, event)
+	} else {
+		err = fmt.Errorf("test err")
+	}
+	return res, err
+}
+
+func (m *MockEventService) Find(_ context.Context, id uint) (*core.Event, error) {
+	var res *core.Event
+	var err error
+	if id == 0 {
+		res = &core.Event{
+			ID:           0,
+			CreatedAt:    time.Time{},
+			UpdatedAt:    time.Time{},
+			DeletedAt:    nil,
+			Experiment:   "testExperiment",
+			Namespace:    "testNamespace",
+			Kind:         "testKind",
+			Message:      "test",
+			StartTime:    nil,
+			FinishTime:   nil,
+			Duration:     "1m",
+			Pods:         nil,
+			ExperimentID: "testUID",
+		}
+	} else {
+		if id == 1 {
+			err = gorm.ErrRecordNotFound
+		} else {
+			err = fmt.Errorf("test err")
+		}
+	}
+	return res, err
+}
+
+func (m *MockEventService) FindByExperimentAndStartTime(context.Context, string, string, *time.Time) (*core.Event, error) {
+	panic("implement me")
+}
+
+func (m *MockEventService) Create(context.Context, *core.Event) error {
+	panic("implement me")
+}
+
+func (m *MockEventService) Update(context.Context, *core.Event) error {
+	panic("implement me")
+}
+
+func (m *MockEventService) DeleteIncompleteEvents(context.Context) error {
+	panic("implement me")
+}
+
+func (m *MockEventService) DeleteByFinishTime(context.Context, time.Duration) error {
+	panic("implement me")
+}
+
+func (m *MockEventService) UpdateIncompleteEvents(context.Context, string, string) error {
+	panic("implement me")
+}
+
+func TestEvent(t *testing.T) {
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "Event Suite")
+}
+
+var _ = Describe("event", func() {
+	var router *gin.Engine
+	BeforeEach(func() {
+		mockes := new(MockEventService)
+
+		s := Service{
+			conf:    nil,
+			kubeCli: nil,
+			archive: nil,
+			event:   mockes,
+		}
+		router = gin.Default()
+		r := router.Group("/api")
+		endpoint := r.Group("/events")
+		endpoint.GET("", s.listEvents)
+		endpoint.GET("/dry", s.listDryEvents)
+		endpoint.GET("/get", s.getEvent)
+	})
+
+	AfterEach(func() {
```

Comment:
If there don't a need for this hook, I think it's better to remove it.

Reply:
![image](https://user-images.githubusercontent.com/16437100/100198340-abae5300-2f36-11eb-8266-be0d61ba68b6.png)

In the other "*_ test.go" , the same method is used, so I moved here directly.

Reply:
It should be "Add any teardown steps that needs to be executed after each test" here, let me change it.

Reply:
Ok. 👌🏽